src/components/Notifications/form/notification.ts

```diff
@@ -0,0 +1,196 @@
+import { SubscriptionMethodOption } from "types/subscription";
+import { HTTPHeaderRow } from "./HTTPHeaderRow";
+
+/**
+ * getNotificationSchema returns the schema and uiSchema for the notification section of subscriptions.
+ *
+ * @param subscriptionMethods - an object containing information about available subscription methods. The available
+ * subscription methods differ between task/version and project subscriptions.
+ */
+export const getNotificationSchema = (
+  subscriptionMethods: SubscriptionMethodOption[]
+): {
+  schema: SpruceFormProps["schema"];
+  uiSchema: SpruceFormProps["uiSchema"];
+} => ({
+  schema: {
+    type: "object" as "object",
+    title: "Choose How to be Notified",
+    required: ["notificationSelect"],
+    properties: {
+      notificationSelect: {
+        type: "string" as "string",
+        title: "Notification Method",
+        oneOf: [
+          ...subscriptionMethods.map((method) => ({
+            type: "string" as "string",
+            title: method.label,
+            enum: [method.value],
+          })),
+        ],
+      },
+    },
+    dependencies: {
+      notificationSelect: {
+        oneOf: [
+          {
+            required: ["jiraCommentInput"],
+            properties: {
+              notificationSelect: {
+                enum: ["jira-comment"],
+              },
+              jiraCommentInput: {
+                type: "string" as "string",
+                title: "Comment on a JIRA Issue",
+                format: "validJiraTicket",
+                minLength: 1,
+              },
+            },
+          },
+          {
+            required: ["slackInput"],
+            properties: {
+              notificationSelect: {
+                enum: ["slack"],
+              },
+              slackInput: {
+                type: "string" as "string",
+                title: "Slack message",
+                format: "validSlack",
+                minLength: 1,
+              },
+            },
+          },
+          {
+            required: ["emailInput"],
+            properties: {
+              notificationSelect: {
+                enum: ["email"],
+              },
+              emailInput: {
+                type: "string" as "string",
+                title: "Email",
+                format: "validEmail",
+                minLength: 1,
+              },
+            },
+          },
+          {
+            properties: {
+              notificationSelect: {
+                enum: ["evergreen-webhook"],
+              },
+              webhookInput: {
+                type: "object" as "object",
+                title: "",
+                required: ["urlInput"],
+                properties: {
+                  urlInput: {
+                    type: "string" as "string",
+                    title: "Webhook URL",
+                    minLength: 1,
+                  },
+                  secretInput: {
+                    type: "string" as "string",
```

Comment:
<img width="462" alt="image" src="https://user-images.githubusercontent.com/13104717/175554339-b69d1ba0-ee90-4856-9c55-3854a2389e0e.png">
Why is this disabled?

Reply:
this field should always have a default / never be editable because it's generated but I'm not really sure how they generate the secret on the legacy UI 🤔

Reply:
I think that would be best 😭 Here's the ticket: [EVG-17181](https://jira.mongodb.org/browse/EVG-17181) and I'll also leave a comment about it in the code